ui/pages/confirmations/components/confirm/info/hooks/useDecodedTransactionData.ts

```diff
@@ -1,20 +1,20 @@
 import { Hex } from '@metamask/utils';
-import { useSelector } from 'react-redux';
 import { TransactionMeta } from '@metamask/transaction-controller';
+
 import {
   AsyncResult,
   useAsyncResult,
 } from '../../../../../../hooks/useAsyncResult';
 import { decodeTransactionData } from '../../../../../../store/actions';
 import { DecodedTransactionDataResponse } from '../../../../../../../shared/types/transaction-decode';
-import { currentConfirmationSelector } from '../../../../selectors';
+import { useConfirmContext } from '../../../../context/confirm';
 
 export function useDecodedTransactionData(): AsyncResult<
   DecodedTransactionDataResponse | undefined
 > {
-  const currentConfirmation = useSelector(currentConfirmationSelector) as
-    | TransactionMeta
-    | undefined;
+  const { currentConfirmation } = useConfirmContext() as unknown as {
```

Comment:
Minor, is it worth using a template on `useConfirmContext` to avoid the casting in future?

Something like:

```
const { currentConfirmation } = useConfirmContext<TransactionMeta>();
```